sys/xtimer/xtimer_core.c

```diff
@@ -176,42 +176,30 @@ static void _add_timer_to_list(xtimer_t **list_head, xtimer_t *timer)
 /**
  * @brief remove a timer from an ordered list of timers
  */
-static int _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
+static void _remove_timer_from_list(xtimer_t **list_head, xtimer_t *timer)
 {
     while (*list_head) {
         if (*list_head == timer) {
             *list_head = timer->next;
-            timer->offset = 0;
-            timer->long_offset = 0;
-            timer->start_time = 0;
             timer->next = NULL;
-            return 1;
         }
         list_head = &((*list_head)->next);
     }
-
-    return 0;
 }
 
 void xtimer_remove(xtimer_t *timer)
 {
-    bool reschedule = false;
-
     /* time sensitive since the target timer can be fired */
     unsigned int state = irq_disable();
-    if (timer == timer_list_head) {
-        reschedule = true;
-    }
+    timer->offset = 0;
+    timer->long_offset = 0;
```

Comment:
this line in combination with the condition below is probably not what you want ;)